src/ZfrCors/Factory/CorsServiceFactory.php

```diff
@@ -35,11 +36,20 @@ class CorsServiceFactory implements FactoryInterface
      * {@inheritDoc}
      * @return CorsService
      */
-    public function createService(ServiceLocatorInterface $serviceLocator)
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
     {
         /* @var $corsOptions CorsOptions */
-        $corsOptions = $serviceLocator->get('ZfrCors\Options\CorsOptions');
+        $corsOptions = $container->get('ZfrCors\Options\CorsOptions');
 
         return new CorsService($corsOptions);
     }
+    /**
+     * {@inheritDoc}
+     *
+     * @return CorsService
+     */
+    public function createService(ServiceLocatorInterface $container)
```

Comment:
> Remove Factorinterface from the class and just use __invoke instead.

@macnibblet this would be be BC break.
> Sm 2.7 provides compatibility with __invoke?

@bakura10  No. At least I don't know this and this is not mentioned in migration guide,  also tested with @macnibblet suggestion as I thought does not work.